00XX-common-ledger-protocol/00XX-common-ledger-protocol.md

```diff
@@ -0,0 +1,348 @@
+
+### CLP Type IDs
+
+| ID | Type | Request/Response |
+|:--|:--|:--|
+| 1 | `Ack` | Response |
+| 2 | `Response` | Response |
+| 3 | `Error` | Response |
+| 4 | `Prepare` | Request |
+| 5 | `Fulfill` | Request |
+| 6 | `Reject` | Request |
+| 7 | `Message` | Request |
+
+### Sub-Protocol Data Format
+
+```asn1
+ContentType ::= INTEGER {
+  applicationOctetString  (0),
+  textPlainUtf8           (1),
+  applicationJson         (2)
+} (0..255)
+
+ProtocolData ::= SEQUENCE OF SEQUENCE {
+  protocolName IA5String,
+  contentType ContentType,
+  data OCTET STRING
+}
+```
+
+## Flow
+
+CLP uses a simple RPC flow. A request-type CLP packet is sent, and a
+response-type CLP packet is sent in response with the same request ID. The
+request types are `Message`, `Prepare`, `Fulfill` and `Reject`, and the
+response types are `Ack` (which may be [removed]), `Response`, and `Error`.
+
+Because it would be too slow to atomically save all requestIds that are
+processed, they are not idempotent. It is the responsibility of the requestor
+to make sure they don't duplicate requestIds. The implementation should ensure
+that no two in-flight requests are sent out with the same requestId. The
+responder should always send back a response to a request with the same
+requestId.
+
+There are also a couple of tricky cases to handle:
+
+- If an unexpected CLP packet is received, no response should be sent. An unexpected CLP packet is a response for which a request was not sent, or a response for a request which has already been responded to.
+- If an unreadable CLP packet is received, no response should be sent. An unreadable CLP packet is one which is structurally invalid, i.e. terminates before length prefixes dictate or contains illegal characters.
+
+These behaviors are important for preventing accidental feedback loops.  If an
+unexpected packet triggered an error, that error may be unexpected to the
+sender. The sender would reply with another unexpected error, causing an
+infinite loop. Unreadable packets must be ignored too. If an application got
+onto a CLP connection and spoke the wrong protocol, it would trigger an error
+from CLP. This might trigger an error from the application, and it would
+devolve into another infinite loop.
+
+### Message
+
+```asn1
+Message ::= SEQUENCE {
+  protocolData ProtocolData
+}
+```
+
+`Message` is used for sending information to the peer. It contains no
+packet-specific data, only protocol data. [ILQP] packets are attached under the
+protocol name `ilp` with content-type `application/octet-stream`.
+
+- `Response` is returned if the peer acknowledges the `Message`. If the peer
+  has data to send in reply (e.g. a quote response), it is carried in the
+protocol data.
+
+- `Error` is returned if the peer is not able to process the message, or there
```

Comment:
s/message/packet

Reply:
The term message in this context refers to a CLP packet of type `Message`, so I think it is ok to keep it. I changed message to `Message` to be consistent with the other packet type descriptions.